site/docs/hosting/vercel.md

```diff
@@ -0,0 +1,106 @@
+# Hosting: Vercel Serverless Functions
+
+This tutorial will guide you on how to deploy your bot to [Vercel](https://vercel.com/) by using [Vercel Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions), assuming that you already have a [Vercel](https://vercel.com) account.
+
+## Project Structure
+
+The only prerequisite to get started with **Vercel Serverless Functions** is to move your code to the `api/` directory as shown below. You can also see [Vercel's documentation](https://vercel.com/docs/concepts/functions/serverless-functions#deploying-serverless-functions) for more on this.
+
+```asciiart:no-line-numbers
+.
+├── node_modules/
+├── build/
+├── api/
+│   └── bot.ts
+├── package.json
+├── package-lock.json
+└── tsconfig.json
+```
+
+If you are using TypeScript, you might as well want to install `@vercel/node` as a dev dependency, but it is not mandatory to follow this guide.
+
+## Configuring Vercel
+
+The next step is to create a `vercel.json` file at the top level of your project. For our example structure, its content would look be:
+
+```json
+{
+  "functions": {
+    "api/bot.ts": {
+      "memory": 1024,
+      "maxDuration": 10
+    }
+  }
+}
+```
+
+> Vercel's free subscription has restrictions on quotas, which we need to enable in the configuration file, `vercel.json` ([grammY Examples](https://github.com/grammyjs/examples/blob/main/vercel-bot/api/index.ts)).
+
+If you want to learn more about this configuration file, see [its documentation](https://vercel.com/docs/project-configuration).
+
+## Configuring TypeScript
+
+In this case, in `tsconfig.json` we can define our _output directory_, for this case is **build** as well as our root directory is **api**. This is important because we will have to add it in the Vercel deploy options.
+
+```json
+{
+  "compilerOptions": {
+    "target": "es2016",
+    "module": "commonjs",
+    "rootDir": "./api",
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "outDir": "./build",
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "strict": true,
+    "skipLibCheck": true
+  }
+}
+```
+
+## Our bot.ts or bot.js file
+
+Regardless if we are using TypeScript or JavaScript, we should have a file through which our bot runs. The only thing we have to add at the end of it is this line:
+
+```ts
+import { Bot, webhookCallback } from "grammy";
+
+const bot = new Bot(""); // <-- put your authentication token between the ""
```

Comment:
FR?

Reply:
Sorry but I don't understand you. What does FR means?

Reply:
For real? Do you put the token there or inside an environment variable?

Reply:
ok! I got it! I'm using `dotenv` but for this example I just followed [the grammY getting started](https://grammy.dev/guide/getting-started.html), but yeah, it's better to do `const bot = new Bot(process.env.BOT_TOKEN);

Reply:
To be fair, the grammY getting started guide makes the same mistake. This is intentional: it opts for the fastest setup possible, rather than following best practices all the way. This is left to the users for a later point.

This deployment guide is a good point to start using env vars. @roj1512 thanks for pointing it out :)

Reply:
@jgcarrillo Can you modify it to the better way? :)

Reply:
Sure! thanks for the review and for this amazing library. Anything you need to contribute to I'm in!